### ISSUE TITLE
Deadlock in Git via Task.Wait()

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#7363]
Some task is trying to run on the UI thread while the UI thread is waiting on the task.

https://gist.github.com/Therzok/4d38c3425d0ccb5927d0e800ad3bd7fe

```
=== Visual Studio Community 2019 Preview for Mac ===

Version 8.0 Preview (8.0 build 2966)
Installation UUID: 010a6a17-65d7-44f0-9e3d-48e5afe82e81
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.6.0.2 (d16-0 / 040682909)

	Package version: 518010003

=== Mono Framework MDK ===

Runtime:
	Mono 5.18.1.3 (2018-08/fdb26b0a445) (64-bit)
	Package version: 518010003

=== NuGet ===

Version: 4.8.2.5835

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.0.0-preview-27122-01
	2.2.0
	2.1.8
	2.1.7
	2.1.2
SDK: /usr/local/share/dotnet/sdk/3.0.100-preview-009812/Sdks
SDK Versions:
	3.0.100-preview-009812
	2.2.101
	2.2.100
	2.1.504
	2.1.503
	2.1.302
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.18.1/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.9
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 10.0 (14320.25)
Build 10A255

=== Xamarin.Mac ===

Version: 5.6.0.2 (Visual Studio Community)
Hash: 040682909
Branch: d16-0
Build date: 2018-12-14 06:23:20-0500

=== Xamarin.iOS ===

Version: 12.0.0.15 (Visual Studio Community)
Hash: 84552a46
Branch: xcode10
Build date: 2018-09-17 21:54:33-0400

=== Xamarin.Android ===

Version: 9.1.5.1 (Visual Studio Community)
Android SDK: /Users/therzok/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		4.0.3  (API level 15)
		4.1    (API level 16)
		4.2    (API level 17)
		4.3    (API level 18)
		4.4    (API level 19)
		4.4.87 (API level 20)
		5.0    (API level 21)
		5.1    (API level 22)
		6.0    (API level 23)
		7.0    (API level 24)
		7.1    (API level 25)
		8.0    (API level 26)
		8.1    (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.0
SDK Build Tools Version: 27.0.0

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/therzok/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android Device Manager ===

Version: 1.2.0.14
Hash: 86df26f
Branch: remotes/origin/d16-0
Build date: 2019-03-15 21:58:58 UTC

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Xamarin Designer ===

Version: 4.17.4.413
Hash: db42fb8b0
Branch: remotes/origin/d16-0
Build date: 2019-03-20 12:37:12 UTC

=== Build Information ===

Release ID: 800002966
Git revision: c5184b1a8925d08199581855b8ab17afa4ebb0b8
Build date: 2019-03-24 14:07:37+00
Build branch: master
Xamarin extensions: bcb32db03bd78a9858271526bff1784cfc0f1d5b

=== Operating System ===

Mac OS X 10.14.3
Darwin 18.2.0 Darwin Kernel Version 18.2.0
    Thu Dec 20 20:46:53 PST 2018
    root:xnu-4903.241.1~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

AddinMaker 1.5.0
```

> VS bug [#827015](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/827015)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-10] Comment by **xamarinc**



Fixed in version 8.0.0.2972 (release-8.0)



Pull Request #409 merged by: Lluis Sanchez  
 Author: xamarin  
 Commit: ec95ce77b3bad8aef3a2928d888446697e845bac (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-12] Comment by **xamarinc**



Fixed in version 8.1.0.1701 (master-vnext)



Pull Request #408 merged by: Marius Ungureanu  
 Author: xamarin  
 Commit: 7ee1010a42b408a2904e72878c9c507184407d68 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #408 merged by: Marius Ungureanu
Author: xamarin
Commit: 7ee1010a42b408a2904e72878c9c507184407d68 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-22] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #408 merged by: Marius Ungureanu  
 Author: xamarin  
 Commit: 7ee1010a42b408a2904e72878c9c507184407d68 (xamarin/monodevelop)
